### DISCUSSION TITLE
Increase options/guidance for compliance with Web Content Accessibility Guidelines (WCAG) in revealjs HTML output

### DISCUSSION THREAD

**Discussion by rressler** [quarto-dev/quarto-cli#8621] [Feature Requests]
### Description

My organization is requiring strict compliance with WCAG 2.1 for posting any HTML materials or links to other materials. There are at least two snags with current revealjs output that appear to require coding in javascript to resolve. Without it I cannot post html documents or links to them (e.g., on quarto pub) on organization pages.

1. Cannot provide alternative text for logo images on slides. Suggest adding a yaml tag  'logo-alt' to accept alternative text for the 'logo' image.
2. Cannot change the user-scalable attribute to yes to allow zooming in of entire slides, not just elements on a slide. Suggest adding a plug in or a yaml option for user-scalable: yes/no to enable this option.

I am not an expert in WCAG by any means so there may be others that could be addressed (see the[ a11y](https://github.com/marcysutton/reveal-a11y) plugin.

It would also be helpful if there was a plug in or option to convert tabsets to individual pages (_Note that one significant disadvantage to tabsets is that only the first tab will be visible when printing to PDF._) so that HTML pages could be more cleanly converted to PDF for accessibility.

**Comment by mcanouil** [comment]
FYI:
- https://github.com/quarto-dev/quarto-cli/discussions/8592

**Comment by cscheid** [comment]
@rressler are your organization's requirements publicly available someplace we can look? Thanks!

**Comment by rressler** [comment]
https://www.w3.org/WAI/standards-guidelines/wcag/

From: Carlos Scheidegger ***@***.***>
Date: Tuesday, February 6, 2024 at 10:48 AM
To: quarto-dev/quarto-cli ***@***.***>
Cc: Richard Ressler ***@***.***>, Mention ***@***.***>
Subject: Re: [quarto-dev/quarto-cli] Increase options/guidance for compliance with Web Content Accessibility Guidelines (WCAG) in revealjs HTML output (Discussion #8621)

External Email: Use caution with links and attachments.

@rressler<https://urldefense.com/v3/__https:/github.com/rressler__;!!IaT_gp1N!0U4PdYrwi1Ieybnz2D-M39XdZ_ifnHv8TnChfgJ3PyZyDsDwbKCUhoclAON-PMwVfONc5ELXvSCL8Rw25gvyp5ijsaw$> are your organization's requirements publicly available someplace we can look? Thanks!

**Reply by cscheid** [reply]
Sorry, I should have clarified. I know what the WCAGs are - I wanted to know what tools your organization is going to use to enforce them so we know _precisely_ what we're targeting.

**Comment by rressler** [comment]
The enterprise product is Siteimprove. Here is a link to the free browser plug-in for testing<https://www.siteimprove.com/integrations/browser-extensions/>. They use this tool because it is more user-friendly.

Since Quarto is open-source, they offered the following details.

  *   Siteimprove uses its own Alpha Ruleset<https://alfa.siteimprove.com/rules>, which is based on the W3C ACT-Rules<https://act-rules.github.io/rules/>.
  *   Deque Axe<https://www.deque.com/axe/> is another popular tool preferred by developers and uses it’s Axe 4.8<https://dequeuniversity.com/rules/axe/4.8> ruleset which is also based on W3C ACT-Rules.

Both rulesets are extremely similar, so either is fine.

Lastly, automated scanning only covers about 60% of the WCAG 2.1 guidelines<https://www.w3.org/WAI/standards-guidelines/wcag/> so they use manual testing for keyboard access<https://webaim.org/techniques/keyboard/> and Content Scaling<https://webaim.org/resources/evalquickref/#scaling>. These tests only take a few seconds or minutes are a good indicator for testing if content is accessible to more WCAG guidelines.



From: Carlos Scheidegger ***@***.***>
Date: Tuesday, February 6, 2024 at 11:50 AM
To: quarto-dev/quarto-cli ***@***.***>
Cc: Richard Ressler ***@***.***>, Mention ***@***.***>
Subject: Re: [quarto-dev/quarto-cli] Increase options/guidance for compliance with Web Content Accessibility Guidelines (WCAG) in revealjs HTML output (Discussion #8621)

External Email: Use caution with links and attachments.

Sorry, I should have clarified. I know what the WCAGs are - I wanted to know what tools your organization is going to use to enforce them so we know precisely what we're targeting.

**Reply by cscheid** [reply]
Thank you, I appreciate the response - I'll follow up here when we know more.

**Comment by mcanouil** [comment]
I've opened an enhancement issue for `logo-alt` in `revealjs`:
- https://github.com/quarto-dev/quarto-cli/issues/10933
